test/WebJobs.Script.Tests.Integration/Controllers/WarmupScenarios.cs

```diff
@@ -0,0 +1,107 @@
+﻿using Microsoft.Azure.WebJobs.Script.WebHost;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Options;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Net.Http;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace Microsoft.Azure.WebJobs.Script.Tests.Integration.Controllers
+{
+    public class WarmupScenarios
+    {
+        private static readonly TimeSpan SemaphoreWaitTimeout = TimeSpan.FromSeconds(30);
+
+        private TestFunctionHost _testHost;
+        private static SemaphoreSlim _hostSetup;
+        private static SemaphoreSlim _hostBuild;
+
+        public WarmupScenarios()
+        {
+            _hostSetup = new SemaphoreSlim(1, 1);
+            _hostBuild = new SemaphoreSlim(0, 2);
+        }
+
+        [Fact]
+        public async Task TestWarmupEndPoint_WhenHostStarts()
+        {
+            string testScriptPath = @"TestScripts\CSharp";
```

Comment:
Switch to Path.Join

Reply:
`Path.Combine`

Reply:
will update, but isn't `Path.Combine` more relevant if the paths can be relative, or are unknowns, or include directory separators. For two constants to join with a separator, shouldn't `Path.Join` be the targeted approach?

Reply:
Yes. That is correct. Looking at the path above, join would be appropriate.